frontend/views/containers/payments/PaymentsMixin.js

```diff
@@ -0,0 +1,97 @@
+import sbp from '@sbp/sbp'
+import { mapState, mapGetters } from 'vuex'
+import { PAYMENT_COMPLETED } from '@model/contracts/shared/payments/index.js'
+import { simplifyPayment, getPaymentHashes } from '@model/contracts/shared/functions.js'
+import { cloneDeep } from '@model/contracts/shared/giLodash.js'
+
+const PaymentsMixin: Object = {
+  computed: {
+    ...mapState(['currentGroupId']),
+    ...mapGetters([
+      'currentGroupState',
+      'ourUsername',
+      'ourPayments',
+      'paymentsForPeriod',
+      'periodBeforePeriod',
+      'groupPeriodPayments',
+      'currentPaymentPeriod',
+      'paymentHashesForPeriod',
+      'groupIncomeAdjustedDistribution'
+    ])
+  },
+  methods: {
+    async getHistoricalPaymentsInTypes () {
+      const paymentsInTypes = {
+        sent: cloneDeep(this.ourPayments?.sent),
+        received: cloneDeep(this.ourPayments?.received),
```

Comment:
The way these are written suggests that `this.ourPayments` could be `null` or `undefined`. This presents a problem, because if that's true, then `cloneDeep` will return `null`, and an exception will be thrown then these are used later on.

However, looking at the implementation of `ourPayments`, it doesn't seem like this value can be null/undefined, so the `?` should be removed here.

Reply:
I described the implementation of `ourPayments` getter below. It could be return `undefined on the third line.
```
ourPayments (state, getters) {
    const periodPayments = getters.groupPeriodPayments
    if (Object.keys(periodPayments).length === 0) return
    const ourUsername = getters.ourUsername
    const cPeriod = getters.currentPaymentPeriod
    const pPeriod = getters.periodBeforePeriod(cPeriod)
    const currentSent = getters.ourPaymentsSentInPeriod(cPeriod)
    const previousSent = getters.ourPaymentsSentInPeriod(pPeriod)
    const currentReceived = getters.ourPaymentsReceivedInPeriod(cPeriod)
    const previousReceived = getters.ourPaymentsReceivedInPeriod(pPeriod)

    // TODO: take into account pending payments that have been sent but not yet completed
    const todo = () => {
      return getters.groupIncomeAdjustedDistribution.filter(p => p.from === ourUsername)
    }

    return {
      sent: [...currentSent, ...previousSent],
      received: [...currentReceived, ...previousReceived],
      todo: todo()
    }
  }
```
So in my opinion, the `sent` and `received` should be defined this way.
```
sent: cloneDeep(this.ourPayments?.sent || []),
received: cloneDeep(this.ourPayments?.received || []),
```

Reply:
Oh! Good point!! 👍